### DISCUSSION TITLE
Share (a set of) documents by providing a filter link

### DISCUSSION THREAD

**Discussion by torwag** [jonaswinkler/paperless-ng#1441] [Feature Requests]
**Synopsis:**
For many workflows with externals, it would be beneficial to enable access of a set of documents from paperless-ng.

Enable sharing very similar to how nextcloud is doing it, would be of great help, esp. together with the tag and metadata (instead of folder) logic.

**Examples:** 

_Provide all receipts of 2020 to your tax accountant:_

- Create a link (possibly password restricted and time limited) to expose all documents which are tagged "Tax accountant" AND of document type "recipes" AND with the origin date between 1/1/20 - 12/31/21.

_Provide your friends a list of all expense of common travel to share among:_

- Create a link with all documents tagged "Holiday 2020" AND "invoice" or "receipt"

_Provide all documents to an organization to apply for a certain contract:_

- Create a link with password "secret" with all documents tagged "Insurance contract document" limited to "31/12/21" 

_Provide a link to make the most recent documents available to all your coworkers. Make double sure no private stuff is shared:_

- Create a link with password "secret_files" with all documents tagged "coworkers" AND NOT  "private" AND with the origin date not later than "4 weeks". 

**More information:**
The tag and metadata system would allow users, to constantly update the list "on the fly" without stashing the documents into individual folders. That is a common drawback in e.g. nextcloud where people accumulate folders like "Share with Foo", "Share with Bar", etc. over time,  with possibly many duplications of the same documents. Paperless-ng could provide here a much more elegant way.

As an additional bonus, this link could provide an "import into my paperless-ng" button to get all this documents into another system. Hence, enable the exchange between paperless-ng users.

**Comment by TheFehr** [comment]
I think this might not be too complicated. We could probably use the saved views feature as a base for defining the views.
I'm just sadly not very experienced with django.

**Reply by torwag** [reply]
Yes,  I think 70% is there already...  Filtering. What needs to be added is the management of sharing (as you said pretty similar to views) and a specific shared page which can be reached directly by a link (possibly password protected and time limited)
